packages/web-pkg/src/components/AppTemplates/AppWrapper.vue

```diff
@@ -0,0 +1,378 @@
+<template>
+  <main :id="applicationId" class="oc-height-1-1" @keydown.esc="closeApp">
+    <!-- TODO: Figure out what to write into hidden pageTitle for screenreader compatibility -->
+    <!-- <h1 class="oc-invisible-sr" v-text="pageTitle" /> -->
+    <app-top-bar
+      v-if="!loading && !loadingError"
+      :main-actions="fileActions"
+      :resource="resource"
+      @close="closeApp"
+    />
+    <loading-screen v-if="loading" />
+    <error-screen v-else-if="loadingError" :message="loadingError.message" />
+    <div v-else class="oc-height-1-1">
+      <slot v-bind="slotAttrs" />
+    </div>
+  </main>
+</template>
+
+<script lang="ts">
+import {
+  PropType,
+  Ref,
+  defineComponent,
+  onBeforeUnmount,
+  ref,
+  unref,
+  watch,
+  computed,
+  onMounted
+} from 'vue'
+import { DateTime } from 'luxon'
+import { useTask } from 'vue-concurrency'
+import { useGettext } from 'vue3-gettext'
+import { onBeforeRouteLeave } from 'vue-router'
+
+import AppTopBar from 'web-pkg/src/components/AppTopBar.vue'
+import ErrorScreen from 'web-pkg/src/components/AppTemplates/PartialViews/ErrorScreen.vue'
+import LoadingScreen from 'web-pkg/src/components/AppTemplates/PartialViews/LoadingScreen.vue'
+import {
+  UrlForResourceOptions,
+  useAppDefaults,
+  useClientService,
+  useStore
+} from 'web-pkg/src/composables'
+import { Resource } from 'web-client'
+import { DavPermission, DavProperty } from 'web-client/src/webdav/constants'
+import { AppConfigObject } from 'web-pkg/src/apps/types'
+import { Action, ActionOptions } from 'web-pkg/src/composables/actions'
+import { isProjectSpaceResource } from 'web-client/src/helpers'
+import { HttpError } from 'web-pkg/src/errors'
+
+export interface AppWrapperSlotArgs {
+  applicationConfig: AppConfigObject
+  resource: Resource
+  currentContent: Ref<string>
+  isDirty: boolean
+  isReadOnly: boolean
+  url: string
+}
+
+export default defineComponent({
+  name: 'AppWrapper',
+  components: {
+    AppTopBar,
+    ErrorScreen,
+    LoadingScreen
+  },
+  props: {
+    // TODO: Add app-top-bar-actions array prop and pass to AppTopBar
```

Comment:
FIXME: we could do it like `importResourceWithExtension` and pass it to `AppWrapperRoute` or we try to make the component *instance* available and setup return it...